src/ln/msgs.rs

```diff
@@ -563,6 +565,18 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn handle_htlc_fail_channel_update(&self, update: &HTLCFailChannelUpdate);
 }
 
+///A trait to describe the functions required to access the local network topology
+pub trait NetworkTopology : Send + Sync {
+	///This should return a vec of all possible channel announcements or an error
+	fn get_all_channel_announcements(&self)-> Result<Vec<ChannelAnnouncement>, std::fmt::Error>;
+	///This should return a vec of all possible node announcements or an error
+	fn get_all_node_announcements(&self)-> Result<Vec<NodeAnnouncement>, std::fmt::Error>;
```

Comment:
Why have these be separated? Doesn't it make more sense to return a List/Vec of <(ChannelAnnouncement, NodeAnnouncement, ChannelUpdate, ChannelUpdate)> instead?

Reply:
Also boils down SOLID. 
To merge it thou adds extra logic to ensure that we have the same amount of each one thou if we want to sync them up in a tuple. But this should not be a problem as we should have one for each. 

But I dont think BOLT specs require update info or I might be missing it?

Reply:
You should definitely need to also send a channel_update dump so that the receiver learns the feerate info. If the BOLT is missing that it probably needs to be fixed.

Reply:
I added a update channel cache as well to forward